lib/plugins/aws/invokeLocal/index.js

```diff
@@ -104,53 +124,183 @@ class AwsInvokeLocal {
+        Number(this.serverless.service.provider.memorySize) ||
+        1024;
+
+      const lambdaDefaultEnvVars = {
+        LANG: 'en_US.UTF-8',
+        LD_LIBRARY_PATH:
+          '/usr/local/lib64/node-v4.3.x/lib:/lib64:/usr/lib64:/var/runtime:/var/runtime/lib:/var/task:/var/task/lib', // eslint-disable-line max-len
+        LAMBDA_TASK_ROOT: '/var/task',
+        LAMBDA_RUNTIME_DIR: '/var/runtime',
+        AWS_REGION: this.provider.getRegion(),
+        AWS_DEFAULT_REGION: this.provider.getRegion(),
+        AWS_LAMBDA_LOG_GROUP_NAME: this.provider.naming.getLogGroupName(lambdaName),
+        AWS_LAMBDA_LOG_STREAM_NAME: '2016/12/02/[$LATEST]f77ff5e4026c45bda9a9ebcec6bc9cad',
+        AWS_LAMBDA_FUNCTION_NAME: lambdaName,
+        AWS_LAMBDA_FUNCTION_MEMORY_SIZE: memorySize,
+        AWS_LAMBDA_FUNCTION_VERSION: '$LATEST',
+        NODE_PATH: '/var/runtime:/var/task:/var/runtime/node_modules',
+      };
 
-    const credentialEnvVars = {};
-    const { cachedCredentials } = this.provider;
-    if (cachedCredentials) {
-      if (cachedCredentials.accessKeyId) {
-        credentialEnvVars.AWS_ACCESS_KEY_ID = cachedCredentials.accessKeyId;
-      }
-      if (cachedCredentials.secretAccessKey) {
-        credentialEnvVars.AWS_SECRET_ACCESS_KEY = cachedCredentials.secretAccessKey;
+      const credentialEnvVars = {};
+      const { cachedCredentials } = this.provider;
+      if (cachedCredentials) {
+        if (cachedCredentials.accessKeyId) {
+          credentialEnvVars.AWS_ACCESS_KEY_ID = cachedCredentials.accessKeyId;
+        }
+        if (cachedCredentials.secretAccessKey) {
+          credentialEnvVars.AWS_SECRET_ACCESS_KEY = cachedCredentials.secretAccessKey;
+        }
+        if (cachedCredentials.sessionToken) {
+          credentialEnvVars.AWS_SESSION_TOKEN = cachedCredentials.sessionToken;
+        }
       }
-      if (cachedCredentials.sessionToken) {
-        credentialEnvVars.AWS_SESSION_TOKEN = cachedCredentials.sessionToken;
+
+      // profile override from config
+      const profileOverride = this.provider.getProfile();
+      if (profileOverride) {
+        lambdaDefaultEnvVars.AWS_PROFILE = profileOverride;
       }
-    }
 
-    // profile override from config
-    const profileOverride = this.provider.getProfile();
-    if (profileOverride) {
-      lambdaDefaultEnvVars.AWS_PROFILE = profileOverride;
+      const configuredEnvVars = this.getConfiguredEnvVars();
+
+      const promises = _.entries(configuredEnvVars).map(([name, value]) => {
+        return this.resolveResource(value).then(output => {
+          configuredEnvVars[name] = output;
+        });
+      });
+
+      return BbPromise.all(promises).then(() => {
+        Object.assign(process.env, lambdaDefaultEnvVars, credentialEnvVars, configuredEnvVars);
+      });
+    });
+  }
+
+  resolveExports() {
+    return this.resolveExportsImpl();
+  }
+
+  resolveExportsImpl(token = undefined, cfExports = undefined) {
```

Comment:
Setting `undefined` as default value, has same effect as not setting a default at all, so I would remove it.

We can though set `cfExports = []` as default (and remove internal conditional initialization of it)

Also as a recursive function, it doesn't fit well as a class method. I would move it outside, and define as plain function to be called as: `resolveExports(provider, nextToken, cfExports)`.